src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -1170,6 +1179,190 @@ def corner2offset(self, block, corner):
 
         return offset
 
+#===============================================================================
+#
+class SchedulerYarn(AgentSchedulingComponent):
+
+    # FIXME: clarify what can be overloaded by Scheduler classes
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg):
+
+        AgentSchedulingComponent.__init__(self, cfg)
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        try:
+            #-----------------------------------------------------------------------
+            # Find out how many applications you can submit to YARN. And also keep
+            # this check happened to update it accordingly
+
+
+            #if 'rm_ip' not in self._cfg['lrms_info']:
+            #    raise RuntimeError('rm_ip not in lm_info for %s' \
+            #            % (self.name))
+
+            self._log.info('Checking rm_ip %s' % self._cfg['lrms_info']['lm_info']['rm_ip'])
+            self._rm_ip = self._cfg['lrms_info']['lm_info']['rm_ip']
+            self._service_url = self._cfg['lrms_info']['lm_info']['service_url']
+            self._rm_url = self._cfg['lrms_info']['lm_info']['rm_url']
+            self._client_node = self._cfg['lrms_info']['lm_info']['nodename']
+
+            sample_time = rpu.timestamp()
+            yarn_status = ul.urlopen('http://{0}:8088/ws/v1/cluster/scheduler'.format(self._rm_ip))
+
+            yarn_schedul_json = json.loads(yarn_status.read())
+
+            max_num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['maxApplications']
+            num_app = yarn_schedul_json['scheduler']['schedulerInfo']['queues']['queue'][0]['numApplications']
+
+            #-----------------------------------------------------------------------
+            # Find out the cluster's resources
+            cluster_metrics = ul.urlopen('http://{0}:8088/ws/v1/cluster/metrics'.format(self._rm_ip))
+
+            metrics = json.loads(cluster_metrics.read())
+            self._mnum_of_cores = metrics['clusterMetrics']['totalVirtualCores']
+            self._mmem_size = metrics['clusterMetrics']['totalMB']
+            self._num_of_cores = metrics['clusterMetrics']['allocatedVirtualCores']
+            self._mem_size = metrics['clusterMetrics']['allocatedMB']
+
+            self.avail_app = {'apps':max_num_app - num_app,'timestamp':sample_time}
+            self.avail_cores = self._mnum_of_cores - self._num_of_cores
+            self.avail_mem = self._mmem_size - self._mem_size
+
+
+            return True
+        except:
+            return False
```

Comment:
The return value of `_configure` is not evaluated - you may want to let the exception fall through...